core/iwasm/interpreter/wasm.h

```diff
@@ -867,14 +876,59 @@ wasm_dpgo_get_cur_prof_cnt_info(WASMModule *module, uint32 func_idx,
 }
 
 static inline void
-wasm_dpgo_dump_prof_cnt_info(struct WASMProfCntInfo *cnt, char *buf,
-                             uint32 buf_size)
+wasm_dpgo_prof_cnt_info_to_string(struct WASMProfCntInfo *cnt, char *buf,
+                                  uint32 buf_size)
 {
     snprintf(buf, buf_size, "  OP:0x%x,OFFSET:%u,CNT:%u,1ST_IDX:%u",
              cnt->opcode, cnt->offset, cnt->counter_amount,
              cnt->first_counter_idx);
 }
 
+/* search in `prof_cnts_info[i]` with the given `offset` of `struct
+ * WASMProfCntInfo` and return `counter_amount` and `first_counter_idx`
+ */
+static inline bool
+wasm_dpgo_get_prof_cnt_info(WASMModule *module, uint32 func_idx, uint32 offset,
+                            uint32 *counter_amount, uint32 *first_counter_idx)
+{
+    bh_list *func_prof_cnts_info;
+    void *elem;
+
+    func_prof_cnts_info = wasm_dpgo_get_func_prof_cnts_info(module, func_idx);
+    bh_assert(func_prof_cnts_info);
+
+    elem = bh_list_first_elem(func_prof_cnts_info);
+    while (elem) {
+        struct WASMProfCntInfo *cnt_info = (struct WASMProfCntInfo *)elem;
+        if (cnt_info->offset == offset) {
+            *counter_amount = cnt_info->counter_amount;
+            *first_counter_idx = cnt_info->first_counter_idx;
+            return true;
+        }
+        elem = bh_list_elem_next(elem);
+    }
+
+    return false;
+}
+
+static inline void
```

Comment:
Same as above

Reply:
You are right. I will do that in the later code refractory PR.